test/cli/whole-program_test.py

```diff
@@ -214,6 +216,65 @@ def test_suppress_inline_project_builddir_j(tmpdir):
     os.mkdir(build_dir)
     __test_suppress_inline_project(tmpdir, ['-j2', '--cppcheck-build-dir={}'.format(build_dir)])
 
+@pytest.mark.parametrize("jobs,builddir", ((1,False), (1,True), (2,False), (2,True)))
+def test_addon_rerun(tmpdir, jobs, builddir):
+    """Rerun analysis and ensure that misra CTU works; with and without build dir, with -j1 and -j2"""
+    args = [
+        f'-j{jobs}',
+        '--addon=misra',
+        '--enable=style',
+        '--template={id}',
+        'whole-program']
+    if builddir:
+        args.append(f'--cppcheck-build-dir={tmpdir}')
+    _, _, stderr = cppcheck(args, cwd=__script_dir)
+    assert 'misra-c2012-5.8' in stderr
+    _, _, stderr = cppcheck(args, cwd=__script_dir)
+    assert 'misra-c2012-5.8' in stderr
+
+@pytest.mark.parametrize("jobs", (1,2))
+def test_addon_builddir_use_ctuinfo(tmpdir, jobs):
+    """Test that ctu-info files are used when builddir is used"""
+    args = [
+        f'--cppcheck-build-dir={tmpdir}',
+        f'-j{jobs}',
+        '--addon=misra',
+        '--enable=style',
+        '--template={id}',
+        'whole-program']
+    _, _, stderr = cppcheck(args, cwd=__script_dir)
+    assert 'misra-c2012-5.8' in stderr
+    with open(os.path.join(tmpdir, 'whole1.a1.ctu-info'), 'wt'):
+        pass
+    with open(os.path.join(tmpdir, 'whole2.a1.ctu-info'), 'wt'):
+        pass
+    _, _, stderr = cppcheck(args, cwd=__script_dir)
+    assert 'misra-c2012-5.8' not in stderr
+
+@pytest.mark.parametrize("jobs,builddir", ((1,False), (1,True), (2,False), (2,True)))
+def test_addon_no_artifacts(tmpdir, jobs, builddir):
+    """Test that there are no artifacts left after analysis"""
+    shutil.copyfile(os.path.join(__script_dir, 'whole-program', 'whole1.c'), os.path.join(tmpdir, 'whole1.c'))
```

Comment:
because there can be artifacts in the test folder before I start the test. Want to have a clean folder with just the test files. This test is also executed locally and locally you might have artifacts..

Reply:
That seems unexpected.

#6787 is about detecting such leftovers.

Reply:
But #6787 only works in the CI. what if I add a file locally one way or another in the test folder. For instance by running cppcheck and pressing Ctrl+C while there is some temporary dump file or something.
That is not tested in the CI.

Reply:
Yes, but it is a start (weirdly it doesn't fail). But so we at least know that it works as expected in the default (i.e. no fatal errors/interruption) case. We can move forward from that.

Reply:
> Yes, but it is a start (weirdly it doesn't fail).

sure. I don't disapprove that PR of course.